File: algorithm/src/main/java/baekjoon/p04153/Main.java
package baekjoon.p04153;

import java.util.Arrays;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (true) {
            int[] inputs = intArray(sc.nextLine().split(" "));
            if (all(inputs, 0)) {
                break;
            }

            System.out.println(solve(inputs));
        }
    }

    private static int[] intArray(String[] arr) {
        int[] ret = new int[arr.length];
        for (int i = 0; i < ret.length; i++) {
            ret[i] = Integer.parseInt(arr[i]);
        }

        return ret;
    }

    private static boolean all(int[] arr, int check) {
        for (int n : arr) {
            if (n != check) {
                return false;
            }
        }

        return true;
    }

    private static String solve(int[] inputs) {
        Arrays.sort(inputs);
        if (inputs[0] * inputs[0] + inputs[1] * inputs[1] == inputs[2] * inputs[2]) {
            return "right";
        } else {
            return "wrong";
        }
    }

}

File: algorithm/src/main/java/baekjoon/p04153/Main.kt
package baekjoon.p04153

fun main(args: Array<String>) {
    while (true) {
        val inputs = readLine()!!.split(" ").map { it.toInt() }
        if (inputs.all { it == 0 }) break
        println(solve(inputs))
    }
}

private fun solve(inputs: List<Int>): String {
    val sorted = inputs.sorted()
    val mul = sorted.map { it * it }

    return when (mul[0] + mul[1] == mul[2]) {
        true -> "right"
        false -> "wrong"
    }
}